crates/database/src/trades.rs

```diff
@@ -17,6 +18,51 @@ pub struct TradesQueryRow {
     pub buy_token: Address,
     pub sell_token: Address,
     pub tx_hash: Option<TransactionHash>,
+    pub fee_policies: Vec<FeePolicy>,
+    pub quote: Option<Quote>,
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Quote {
+    pub sell_amount: BigDecimal,
+    pub buy_amount: BigDecimal,
+    pub fee: f64,
+}
+
+impl<'r> sqlx::FromRow<'r, sqlx::postgres::PgRow> for TradesQueryRow {
```

Comment:
isn't it possible to derive this with `sqlx::FromRow`? and then implement `Decode` independently

Reply:
No, it is not possible with this complex query(see https://github.com/cowprotocol/services/pull/2809#discussion_r1675527726).